src/Raven.Server/ServerWide/BackgroundTasks/LatestVersionCheck.cs

```diff
@@ -7,61 +8,69 @@
 using Raven.Server.Json;
 using Raven.Server.NotificationCenter.Notifications;
 using Raven.Server.NotificationCenter.Notifications.Details;
+using Sparrow.Collections;
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Server.ServerWide.BackgroundTasks
 {
-    public class LatestVersionCheck : IDisposable
+    public static class LatestVersionCheck
     {
         private const string ApiRavenDbNet = "https://api.ravendb.net";
 
-        private static readonly Logger _logger = LoggingSource.Instance.GetLogger<LatestVersionCheck>(null);
+        private static readonly Logger _logger = LoggingSource.Instance.GetLogger<ServerStore>(null);
 
-        private Timer _latestVersionCheckTimer;
+        private static AlertRaised _alert;
 
-        private readonly ServerStore _serverStore;
+        private static Timer _timer;
 
-        private static readonly HttpClient ApiRavenDbClient = new HttpClient()
+        private static readonly ConcurrentSet<WeakReference<ServerStore>> ServerStores = new ConcurrentSet<WeakReference<ServerStore>>();
+
+        private static readonly HttpClient ApiRavenDbClient = new HttpClient
         {
             BaseAddress = new Uri(ApiRavenDbNet)
         };
 
-        public LatestVersionCheck(ServerStore serverStore)
+        static LatestVersionCheck()
         {
-            _serverStore = serverStore;
+            _timer = new Timer(state =>
+                AsyncHelpers.RunSync(PerformAsync), null, (int)TimeSpan.FromMinutes(5).TotalMilliseconds, (int)TimeSpan.FromHours(12).TotalMilliseconds);
```

Comment:
We won't have so many of those, sure, but we don't actually need it to run sync.